pkg/blockstorage/azure/client.go

```diff
@@ -118,12 +116,12 @@ func getAuthorizer(env azure.Environment, config map[string]string) (*autorest.B
 	}
 
 	credConfig := auth.NewClientCredentialsConfig(clientID, clientSecret, tenantID)
-	if aDDEndpoint, ok := config[blockstorage.AzureActiveDirEndpoint]; ok {
-		credConfig.AADEndpoint = aDDEndpoint
+	if credConfig.AADEndpoint, ok = config[blockstorage.AzureActiveDirEndpoint]; !ok {
```

Comment:
Can we add unit tests to verify this logic where we default to env.Something if ok is false, and verify that we use the config passed in config[some field] when ok is true ?

Reply:
Yea I'm trying to find a way to unit test this